lib/cpp/src/thrift/concurrency/ThreadManager.cpp

```diff
@@ -339,103 +357,88 @@ void ThreadManager::Impl::addWorker(size_t value) {
     idMap_.insert(std::pair<const Thread::id_t, shared_ptr<Thread> >((*ix)->getId(), *ix));
```

Comment:
This means canSleep() will always return true, which is correct; when adding a task without a timeout you will get TooManyPendingTasksException.  When adding a task with a timeout you may get a TimedOutException.